packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -25,6 +24,39 @@ type ProxyPreviewModeHandlerArgs =
 const routeHandlerPattern = [P.instanceOf(Request), P.any, P.any] as const
 const apiRoutePattern = [P.any, P.any, P.any] as const
 
+const previewData: MakeswiftPreviewData = {
+  makeswift: true,
+  siteVersion: MakeswiftSiteVersion.Working,
+}
+
+const mapRequestHeadersToHeaders = (requestHeaders: NextApiRequest['headers']) => {
+  const headers = new Headers()
+
+  Object.entries(requestHeaders).forEach(([key, value]) => {
+    match(value)
+      .with(P.array(P.string), value => headers.set(key, value.join(',')))
```

Comment:
In this case, we'd want to do `headers.append` which is the API for headers with multiple values.